packages/cepcsw/package.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install cepcsw
+#
+# You can edit this file again by typing:
+#
+#     spack edit cepcsw
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
+from spack import *
+
+
+class Cepcsw(CMakePackage):
+    """FIXME: Put a proper description of your package here."""
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepage = "https://github.com/cepc/CEPCSW"
+    url      = "https://github.com/cepc/CEPCSW/archive/master.zip"
+    git      = "https://github.com/cepc/CEPCSW.git"
+
+    # FIXME: Add a list of GitHub accounts to
+    # notify when the package is updated.
+    maintainers = ['mirguest']
+
+
+    variant('cxxstd',
+            default='17',
+            values=('14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+    # FIXME: Add proper versions here.
+    # version('1.2.4')
+    version('master', branch='master')
+
+    # FIXME: Add dependencies if required.
+    depends_on('clhep')
+    depends_on('dd4hep +geant4')
+    depends_on('edm4hep')
+    depends_on('k4fwcore')
+    depends_on('gaudi')
+    depends_on('gear')
+    depends_on('lcio')
+    depends_on('lccontent')
+    depends_on('pandorasdk')
+    depends_on('pandorapfa')
+    depends_on('podio')
+    depends_on('root')
+
+    def cmake_args(self):
+        args = []
+        # C++ Standard
+        args.append('-DCMAKE_CXX_STANDARD=%s'%self.spec.variants['cxxstd'].value)
+        if self.spec.satisfies('^gaudi@:34.99'):
+            args.append('-DHOST_BINARY_TAG=x86_64-linux-gcc9-opt')
+
+        clhep_prefix = self.spec["clhep"].prefix
```

Comment:
Is this in fact necessary? I think we have some other packages that also use clhep and work fine with the default cmake configuration.

Reply:
Hi @vvolkl , yes, it looks like necessary when I use the key4hep stack. If I don't specify the CLHEP, I will got some errors:
```
1 error found in build log:
     340    -- Found GSL: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/gsl-2.5-5knpqpqmm4pkgstpjszj43n6rdl6xijt/include (found version "2.5")
     341    GSL: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/gsl-2.5-5knpqpqmm4pkgstpjszj43n6rdl6xijt/lib/libgsl.so;/cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/gsl-2.5-5knpqpqm
            m4pkgstpjszj43n6rdl6xijt/lib/libgslcblas.so
     342    GSL INCLUDE_DIRS: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/gsl-2.5-5knpqpqmm4pkgstpjszj43n6rdl6xijt/include
     343    -- Adding directory Utilities/KalTest (12/28)
     344    -- Found ROOT: /tmp/lint/spack/opt/spack/linux-centos7-broadwell/gcc-8.3.0/root-6.20.04-5rydruco6werkg6j3o7q2jr5d25vjygu/include
     345    -- Adding directory Utilities/KalDet (13/28)
  >> 346    CMake Error at /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/cmake-3.18.2-g3i5ynl2wecvrhhhunv3g743mdvou46p/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
     347      Could NOT find CLHEP (missing: CLHEP_INCLUDE_DIR)
     348    Call Stack (most recent call first):
     349      /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/cmake-3.18.2-g3i5ynl2wecvrhhhunv3g743mdvou46p/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
     350      /tmp/lint/spack/opt/spack/linux-centos7-broadwell/gcc-8.3.0/gaudi-34.0-43ds2avyvnlapo7epsqbbvegvmbdkfi6/cmake/modules/FindCLHEP.cmake:64 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
     351      Utilities/KalDet/CMakeLists.txt:8 (find_package)
     352

See build log for details:
  /tmp/lint/spack-stage/spack-stage-cepcsw-master-juud46faax3ahasz7e4weicuku5uflpk/spack-build-out.txt
```

Maybe the CMakeLists.txt of some packages in CEPCSW is not well written. 
I will try to remove them when it's fixed.

Tao

Reply:
Not sure what happens for the other packages, but ideally one should avoid picking up the FindXYZ.cmake modules from Gaudi. 

>      350      /tmp/lint/spack/opt/spack/linux-centos7-broadwell/gcc-8.3.0/gaudi-34.0-43ds2avyvnlapo7epsqbbvegvmbdkfi6/cmake/modules/FindCLHEP.cmake:64 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)

CMake itself contains a better FindCLHEP.cmake.

Reply:
Hi @andresailer ,
Thanks. It looks work after I remove the `find_package(CLHEP)`. I will update this receipt after CEPCSW is updated.

Reply:
A better solution would be using `find_package` with option `CONFIG`.